com.avaloq.tools.ddk.typesystem/src/com/avaloq/tools/ddk/typesystem/AbstractTypeProvider.java

```diff
@@ -38,12 +45,19 @@
  *
  * @see org.eclipse.xtext.xbase.typing.AbstractTypeProvider
  */
-public abstract class AbstractTypeProvider implements ITypeProvider {
+public abstract class AbstractTypeProvider implements ITypeProvider, ILanguageFeatureExtension {

```

Comment:
Not in this case because the type provider was already designed to delegate, and the "provides for thing" methods are not dependent on the "get provider for" method, which means we can ask whether it's the right provider without getting caught in a loop.

Reply:
I do have rather concerns about using extensions to contribute type providers. I think we are misusing inference framework and it becomes more and more complex. If the infered object requires different set of injectors than the resource from where inference is triggered, shoudln't it rather be a different resoruce(or smtg else which is not part of the original resource)? 
I think it make sence to discuss this with Roman and Knut.
Reagrding you workflow change, is it possible to infer avaloq script parameters instead wfcl one? Then you would not need an extension for type provider.